Userland/Libraries/LibWeb/ServiceWorker/Job.cpp

```diff
@@ -401,7 +401,7 @@ static void update(JS::VM& vm, JS::NonnullGCPtr<Job> job)
         // 16. If runResult is failure or an abrupt completion, then:
         // 17. Else, invoke Install algorithm with job, worker, and registration as its arguments.
         if (job->client) {
-            auto& realm = *vm.current_realm();
+            auto& realm = job->client->realm();
```

Comment:
I don't know what about the behavioral difference between the two realms, but I think it should also be fixed if you move the TemporaryExecutionContext a line above. It looks like `unregister` also has a similar issue as it is being invoked asynchronously and uses current_realm (which assumes an execution context)

Reply:
Oh, wait, I'm being dumb about moving it a line up, but my point still stands about same boog in unregister 😅

Reply:
Gotcha, and nice catch.